core/src/main/java/org/elasticsearch/rest/action/RestToXContentListener.java

```diff
@@ -42,12 +42,20 @@ public final RestResponse buildResponse(Response response) throws Exception {
     }
 
     public final RestResponse buildResponse(Response response, XContentBuilder builder) throws Exception {
-        builder.startObject();
+        if (wrapInObject()) {
```

Comment:
I would love to get rid of those but the default here is starting an object externally, which is conceptually wrong. With this change there is one less of those situations. I expect that we clean this up as we go. objects should be started within toXContent as much as possible, otherwise the writing and parsing code will not be symmetric.